pinot-hadoop-filesystem/src/main/java/com/linkedin/pinot/filesystem/HadoopPinotFS.java

```diff
@@ -121,7 +123,32 @@ public long length(URI fileUri) throws IOException {
 
   @Override
   public void copyToLocalFile(URI srcUri, File dstFile) throws Exception {
-    hadoopFS.copyToLocalFile(new Path(srcUri), new Path(dstFile.toURI()));
+    LOGGER.debug("starting to fetch segment from hdfs");
+    final String dstFilePath = dstFile.getAbsolutePath();
+    try {
+      final Path remoteFile = new Path(srcUri);
+      final Path localFile = new Path(dstFile.toURI());
+
+      RetryPolicy fixDelayRetryPolicy = RetryPolicies.fixedDelayRetryPolicy(retryCount, retryWaitMs);
+      fixDelayRetryPolicy.attempt(() -> {
+        try {
+          if (hadoopFS == null) {
```

Comment:
I guess this can never be null after your fix?

Reply:
could be if we instantiate the hadoop one, not the local one and the instantiation fails

      _hadoopFS = org.apache.hadoop.fs.FileSystem.get(_hadoopConf);